### ISSUE TITLE
error _buffer_to_frame

### ISSUE THREAD

**Issue by MarcelMB** [Aharoni-Lab/miniscope-io#34]
![Screenshot 2024-07-11 at 4 47 40 PM (1)](https://github.com/user-attachments/assets/8e69e982-0a42-493e-a4a1-27876e8dbd47)

Running into this error for the newest release on main [f11d901](https://github.com/Aharoni-Lab/miniscope-io/commit/f11d9018e8bde5587664c0575bc51cec2a7e1892)

Had already talked with @t-sasatani about it. decided to get the full binary output with:
 `mio stream capture -c /Users/mbrosch/Documents/GitKraken_mac/miniscope-io/miniscope_io/data/config/WLMS_v02_200px.yml -b binary_error_test.bin`

[Download the binary file](https://1drv.ms/u/s!AmjvNbQEXMGfhIF_WGEAuy83DLqdZA?e=raNa7Q)

**Comment by sneakers-the-rat** [comment]
Off by 1 error? Ill take a look in a minute

**Comment by t-sasatani** [comment]
Yes, this itself seems like it can be patched by adding one or two error handling. I wanted it here because we might want to look closely at how a (presumably) broken buffer generally affects image construction and consider whether we're doing the right amount of carefulness.

**Comment by t-sasatani** [comment]
To dos
- [ ] Patch this with some error handling.
- [ ] Think and implement how to recover data even if some metadata is corrupted. Some stuff to start with:
    - Assume the metadata won't do crazy changes and complement them under specific conditions.
    - Error correction code for metadata.
